### PR TITLE
chore: add matchers tests

### PR DESCRIPTION
**Pull request by eolatham**
This completes [matchers_test.go](https://github.com/guidewire-oss/sawchain/blob/main/matchers_test.go) with tests for `MatchYAML` and `HaveStatusCondition`.
Part of https://github.com/guidewire-oss/sawchain/issues/1.
    

---

## Summary by cubic
Added comprehensive tests for the MatchYAML and HaveStatusCondition matchers to cover success, failure, and edge cases.

- **Tests**
  - Added table-driven tests for MatchYAML with typed and unstructured objects, bindings, and error scenarios.
  - Added table-driven tests for HaveStatusCondition with various condition types, statuses, and error cases.
  - Removed outdated comments from test suite files.

### CODE REVIEW

File: suite_test.go [hunk 1]
```diff
@@ -20,8 +20,12 @@ const (
 	fastInterval = 5 * time.Millisecond
 )
 
-// Variables must be assigned inline to beat static Entry parsing!
-var ctx = context.Background()
+var (
+	ctx = context.Background()
+
+	standardClient         = testutil.NewStandardFakeClient()
```
[the remark this answers is not in the document]
Comment (eolatham) [reply suite_test.go]: This has been addressed by moving the client variables to the Describe where they are used. Reuse within the describe is okay because no actual CRUD operations are done in these particular tests.
